ground/gcs/src/plugins/config/outputchannelform.cpp

```diff
@@ -387,9 +387,19 @@ void OutputChannelForm::updateMaxSpinboxValue(UAVObject *obj)
             // ... and if there's a match, set the maximum values and return
             if (channel-1 == m_index) {
                 if (actuatorSettingsData.TimerUpdateFreq[i] == 0)
+                {
+                    // SyncPWM has been selected, limit values to maximum allowed by UAVO
+                    ui.actuatorMin->setMaximum(std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max());
+                    ui.actuatorMax->setMaximum(std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max());
                     return;
-
-                double maxPulseWidth = round(10000000.0 / actuatorSettingsData.TimerUpdateFreq[i]);
+                }
+
+                double pwm_freq = 0;
+                if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_1MHZ)
+                    pwm_freq = 1000000;
+                else if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_12MHZ)
+                    pwm_freq = 12000000;
+                double maxPulseWidth = round(pwm_freq / actuatorSettingsData.TimerUpdateFreq[i]);
```

Comment:
This looks like wrong math. The pulse width is now in true microseconds and the resolution doesn't matter (since we round to microseconds, untill someone adds a mode with a resolution of 10s of ms.

The right math would be to always ```round(pwm_freq / actuatorSettingsData.TimerUpdateFreq[i])```. However, if ```actuatorSettingsData.TimerUpdateFreq==0``` which indicates SyncPWM then this code produces a divide by 0 error. Here, the maximum pulse duration is a function of the sensor period. Most sensors are running between 300-500 Hz, some people go as high as 1khz. It might be appropriate to hardcode the max to 1000us for SyncPWM to cover all these cases robustly...